library/nxos_switchport

```diff
@@ -116,16 +130,25 @@ options:
 '''
 EXAMPLES = '''
 # Config a switchport to be a trunk interface with native vlan 10 and carriers vlan 2-100
-- nxos_switchport: interface=Ethernet1/1 mode=trunk native_vlan=10 trunk_vlans=2-100 host={{ inventory_hostname }}
+- nxos_switchport: interface=Ethernet1/1 mode=trunk native_vlan=10 trunk_vlans=2-100 host={{ inventory_hostname }} state=present
+
+# Config a switchport to be a trunk interface and add vlan 101-120 as allowed
+- nxos_switchport: interface=Ethernet1/1 mode=trunk trunk_vlans=101-120 host={{ inventory_hostname }} state=present
 
 # Config a switchport to an access port on vlan 20
-- nxos_switchport: interface=Ethernet1/2 mode=access access_vlan=20 host={{ inventory_hostname }}
+- nxos_switchport: interface=Ethernet1/2 mode=access access_vlan=20 host={{ inventory_hostname }} state=present
 
 # Remove existing access port vlan configuration on a switchport (mode is required)
 - nxos_switchport: interface=Ethernet1/2 host={{ inventory_hostname }} mode=access state=absent
 
-# Remove existing trunk port vlan configuration on a switchport (mode is required)
+# Modify existing trunk port vlan configuration on a switchport (mode is required) and remove all vlans
 - nxos_switchport: interface=Ethernet1/1 host={{ inventory_hostname }} mode=trunk state=absent
+
+# Modify existing trunk port vlan configuration on a switchport (mode is required) and remove vlan 1-100
+- nxos_switchport: interface=Ethernet1/1 host={{ inventory_hostname }} mode=trunk trunk_vlans=1-100 state=absent
+
+# Set switchport to default configuration
+- nxos_switchport: interface=Ethernet1/2 host={{ inventory_hostname }} state=default
```

Comment:
`mode` should still be required.  So, technically it's the default configuration for the given switchport type (assumes admin knows it). If they don't, they should use nxos_interface to **really** default the interface.